app/validators/all_details_on_order_validator.rb

```diff
@@ -0,0 +1,39 @@
+class AllDetailsOnOrderValidator
+
+  cattr_accessor :order_detail_validator_class
+
+  # Build a validator that will take an Order in its constructor and validate each
+  # child order detail using the order_detail_validator_class.
+  #
+  # Example:
+  # validator = AllDetailsOnOrder.build(NotePresenceValidator)
+  # validator.new(order).valid?
+  #
+  # This will run NotePresenceValidator.new(order_detail).valid? for each
+  # element of @order.order_details
+  def self.build(order_detail_validator_class)
+    Class.new(AllDetailsOnOrderValidator) do
+      self.order_detail_validator_class = order_detail_validator_class
+    end
+  end
+
+  # This class should only be initialized through the `.build` method
+  def initialize(order)
+    raise "Should not be initialized directly. Use the `.build` factory method instead" unless order_detail_validator_class
+    @order = order
+  end
+
+  def valid?
+    # Use select instead of none? to make sure we loop over everything
```

Comment:
Ah I see what you're saying. I would then amend the comment to include why we want to make sure to loop over all the validators, i.e. we want to collect all the errors.